src/python/tests/core/bot/tasks/component_related_test.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for blame task."""
+# pylint: disable=protected-access
+
+import ast
+import json
+import os
+import unittest
+
+from bot.tasks import blame_task
+from datastore import data_handler
+from system import environment
+from tests.test_libs import helpers
+from tests.test_libs import test_utils
+
+DATA_DIRECTORY = os.path.join(os.path.dirname(__file__), 'component_related_test_data')
+
+
+@test_utils.with_cloud_emulators('datastore')
+class ComponentRelatedTest(unittest.TestCase):
```

Comment:
Nit: This and the filename seem a bit hard to make sense of without the context of this CL. Could you rename them to something a bit more descriptive? Maybe something along the lines of ComponentRevisionPatchingTest?